### DISCUSSION TITLE
Is it possible to recover lost work after a crash?

### DISCUSSION THREAD

**Discussion by sqwxl** [helix-editor/helix#4217] [Q&A]
```
thread 'main' panicked at 'Position 3824 is out of range for changeset len 3571!', helix-core/src/transaction.rs:397:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This happened and and upon re-opening the file I was working on, all my latest changes were gone (I forgot to save). With vim/neovim I am used to the program keeping a swapfile around for these types of scenarios, but there doesn't seem to be the same functionality in helix. Is there any way I can recuperate my work, or is it irretrievable?

Concerning the error itself, I wish I could give steps on how to reproduce it, but unfortunately it was caused by a typo while I was inputting a motion and I don't know what I typed...

**Comment by Omnikar** [comment]
See #290
